### PR TITLE
Remote certificate loading

### PR DESCRIPTION
**Pull request by grittygrease**
- New OpCode for fetching a certificate: 0x20
- New tag for carrying a list of sigAlgs: TagSigAlgs (0x06)
- Callback function for returning a certificate given SNI, ClientIP,
  SigAlgs

### CODE REVIEW

**Comment by elithrar** [comment]
LGTM.

File: server/server.go [hunk 1]
```diff
@@ -235,6 +240,22 @@ func (s *Server) handle(conn *gokeyless.Conn) {
 			s.stats.logRequest(requestBegin)
 			continue
 
+		case gokeyless.OpCertificateRequest:
+			if s.CertLoader == nil {
+				log.Error(gokeyless.ErrKeyNotFound)
```
Comment (lziest) [inline server/server.go]: Probably not KeyNotFound

Comment (grittygrease) [inline server/server.go]: Added a new error

**Comment by lziest** [comment]
LGTM

File: tests/client_test.go [hunk 2]
```diff
@@ -145,3 +145,27 @@ func TestRSADecrypt(t *testing.T) {
 		t.Fatal("rsa decrypt suceeded despite incorrect SessionKeyLen")
 	}
 }
+
+func TestCertLoad(t *testing.T) {
+	certChainBytes := []byte("DUMMYCHAIN")
```
Comment (lziest) [inline tests/client_test.go]: need to update this for test to pass.

Comment (grittygrease) [inline tests/client_test.go]: fixed!

File: client/client.go [hunk 3]
```diff
@@ -332,3 +332,68 @@ func (c *Client) LoadTLSCertificate(server, certFile string) (cert tls.Certifica
...
+		}
+
+		for {
+			certDERBlock, certPEMBlock = pem.Decode(certPEMBlock)
+			if certDERBlock == nil {
+				break
+			}
+
+			if certDERBlock.Type == "CERTIFICATE" {
+				cert.Certificate = append(cert.Certificate, certDERBlock.Bytes)
```
Comment (kisom) [inline client/client.go]: Where is `cert` setup? I think it's `nil` here, and this will panic.

Comment (grittygrease) [inline client/client.go]: Thanks, this is fixed now and I added a testcase to catch it.

**Comment by kisom** [comment]
LGTM.

**Pull request merged**